iris-mpc-gpu/src/dot/distance_comparator.rs

```diff
@@ -347,4 +408,16 @@ impl DistanceComparator {
             })
             .collect::<Vec<_>>()
     }
+
+    pub fn prepare_luc_bitmap(&self, db_size: usize) -> Vec<CudaSlice<u64>> {
+        let row_stride64 = (db_size + 63) / 64;
+        (0..self.device_manager.device_count())
+            .map(|i| {
+                self.device_manager
+                    .device(i)
+                    .alloc_zeros(row_stride64 * self.query_length)
```

Comment:
why multiply by `self.query_length` here?

Reply:
As discussed: we allocate a 2D array, flattened to a 1D bitmap for each of the queries, as each query will have (in the future) its own set of serial ids against which the OR ruling is used

Reply:
as discussed, this should be divided by `ROTATIONS` because `query_length` is multiplied by it: https://github.com/worldcoin/iris-mpc/blob/f4dbe47c9beaf2806369ee0bec5ffb60b4c9d222/iris-mpc-gpu/src/server/actor.rs#L221